x-pack/plugins/cases/public/containers/use_get_feature_ids.test.tsx

```diff
@@ -6,65 +6,72 @@
  */
 
 import { renderHook, act } from '@testing-library/react-hooks';
-import React from 'react';
-import { TestProviders } from '../common/mock';
+import { useToasts } from '../common/lib/kibana';
+import type { AppMockRenderer } from '../common/mock';
+import { createAppMockRenderer } from '../common/mock';
 import { useGetFeatureIds } from './use_get_feature_ids';
 import * as api from './api';
 
 jest.mock('./api');
 jest.mock('../common/lib/kibana');
 
 describe('useGetFeaturesIds', () => {
+  const addSuccess = jest.fn();
+  const addError = jest.fn();
+
+  (useToasts as jest.Mock).mockReturnValue({ addSuccess, addError });
+
+  let appMockRender: AppMockRenderer;
+
   beforeEach(() => {
+    appMockRender = createAppMockRenderer();
     jest.clearAllMocks();
-    jest.restoreAllMocks();
   });
 
   it('inits with empty data', async () => {
-    jest.spyOn(api, 'getFeatureIds').mockRejectedValue([]);
-    const { result } = renderHook(() => useGetFeatureIds(['context1']), {
-      wrapper: ({ children }) => <TestProviders>{children}</TestProviders>,
+    const spy = jest.spyOn(api, 'getFeatureIds').mockRejectedValue([]);
+    const { waitForNextUpdate } = renderHook(() => useGetFeatureIds(['context1']), {
+      wrapper: appMockRender.AppWrapper,
     });
 
-    await act(async () => {
-      expect(result.current.alertFeatureIds).toEqual([]);
-      expect(result.current.isLoading).toEqual(true);
-      expect(result.current.isError).toEqual(false);
-    });
+    await waitForNextUpdate();
+    expect(spy).toHaveBeenCalledWith(
+      { registrationContext: ['context1'] },
+      expect.any(AbortSignal)
+    );
   });
 
   it('fetches data and returns it correctly', async () => {
     const spy = jest.spyOn(api, 'getFeatureIds');
-    const { result } = renderHook(() => useGetFeatureIds(['context1']), {
-      wrapper: ({ children }) => <TestProviders>{children}</TestProviders>,
+    const { waitForNextUpdate } = renderHook(() => useGetFeatureIds(['context1']), {
```

Comment:
I think this test is not needed. It is the same as the first one. Maybe we can delete it and rename the first test from `inits with empty data` to `returns the features ids correctly`